interface/Tools.h

```diff
@@ -74,24 +76,296 @@ MELAAngles HHAnalyzer::getMELAAngles(const LorentzVector &q1, const LorentzVecto
+                if (std::find(hlt.object_filters[i2].begin(), hlt.object_filters[i2].end(), f1) != hlt.object_filters[i2].end())
+                    leptons[dilepton.ilep2].hlt_leg1 = true;
+        for (auto& i2: l2_samepath_indices)
+            for (auto& f2: filter_leg2)
+                if (std::find(hlt.object_filters[i2].begin(), hlt.object_filters[i2].end(), f2) != hlt.object_filters[i2].end())
+                    leptons[dilepton.ilep2].hlt_leg2 = true;
+    } else if (leptons[dilepton.ilep1].isEl && leptons[dilepton.ilep2].isEl) {
+        // di-electron filters: from the path name the legs have symetric cuts
+        // meaning: set everything to true, leg1 and leg2 are determined by the highest hlt pt
+        if (HH_HLT_DEBUG) std::cout << "\tfinding dilepton legs: di-electron" << std::endl;
+        filter_leg1.push_back("hltEle17Ele12CaloIdLTrackIdLIsoVLTrackIsoLeg1Filter");
+        filter_leg1.push_back("hltEle23Ele12CaloIdLTrackIdLIsoVLTrackIsoLeg1Filter");
+        filter_leg2.push_back("hltEle17Ele12CaloIdLTrackIdLIsoVLTrackIsoLeg2Filter");
+        filter_leg2.push_back("hltEle23Ele12CaloIdLTrackIdLIsoVLTrackIsoLeg2Filter");
+        for (auto& i1: l1_samepath_indices)
+            for (auto& f1: filter_leg1)
+                if (std::find(hlt.object_filters[i1].begin(), hlt.object_filters[i1].end(), f1) != hlt.object_filters[i1].end())
+                    leptons[dilepton.ilep1].hlt_leg1 = true;
+        for (auto& i1: l1_samepath_indices)
+            for (auto& f2: filter_leg2)
+                if (std::find(hlt.object_filters[i1].begin(), hlt.object_filters[i1].end(), f2) != hlt.object_filters[i1].end())
+                    leptons[dilepton.ilep1].hlt_leg2 = true;
+        for (auto& i2: l2_samepath_indices)
+            for (auto& f1: filter_leg1)
+                if (std::find(hlt.object_filters[i2].begin(), hlt.object_filters[i2].end(), f1) != hlt.object_filters[i2].end())
+                    leptons[dilepton.ilep2].hlt_leg1 = true;
+        for (auto& i2: l2_samepath_indices)
+            for (auto& f2: filter_leg2)
+                if (std::find(hlt.object_filters[i2].begin(), hlt.object_filters[i2].end(), f2) != hlt.object_filters[i2].end())
+                    leptons[dilepton.ilep2].hlt_leg2 = true;
+    } else if ((leptons[dilepton.ilep1].isEl && leptons[dilepton.ilep2].isMu)
+        || (leptons[dilepton.ilep1].isMu && leptons[dilepton.ilep2].isEl)) {
+        // if the two offline objects are matching the same different flavour HLT path
+        // then the leg1 and leg2 assignment is in sync with the order of the path name itself
+        if (HH_HLT_DEBUG) std::cout << "\tfinding dilepton legs: different flavour" << std::endl;
+        std::string tmp_mu = "_Mu";
+        std::string tmp_ele = "_Ele";
+        for (auto& i1: l1_samepath_indices) {
+            for (auto& path: hlt.object_paths[i1]) {
+                bool muel = path.find(tmp_mu) < path.find(tmp_ele);
+                if (muel) {
```

Comment:
my bad, actually this is wrong, the cross-trigger path names have always `_Mu` before `_Ele`, this need to be fixed...